### PR TITLE
feat: feature to turn bluetooth on

### LINKED ISSUE #584: On Transfer option select
**Describe the feature you'd like**
 When a user is on the saved badge and he chooses the transfer option the user can easily turn on the Bluetooth within the app. currently, this action shows a toast message Requesting user to turn Bluetooth on.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by refactor-droidyy**
Fixes #584 

Screenshots for the change:

bluetooth

### CODE REVIEW

**Comment by refactor-droidyy** [comment]
@adityastic please review.

**Comment by refactor-droidyy** [comment]
@adityastic Please review.

createBadgeBluetooth


savedBadgeBluetooth

File: app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt [hunk 1]
```diff
@@ -77,6 +77,7 @@ class TextArtFragment : BaseFragment() {
     private val modeAdapter = ModeAdapter()
 
     private val viewModel by sharedViewModel<TextArtViewModel>()
+    private val bluetoothManager: org.fossasia.badgemagic.util.BluetoothManager by inject()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt]: Why using FQN

File: app/src/main/java/org/fossasia/badgemagic/util/BluetoothManager.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package org.fossasia.badgemagic.util
+
+import android.bluetooth.BluetoothManager
+import android.content.Context
+
+class BluetoothManager(val context: Context) {
+    private val btManager = context.getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
+    val btAdapter = btManager.adapter!!
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/badgemagic/util/BluetoothManager.kt]: Keep this private

File: app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt [hunk 3]
```diff
@@ -205,10 +207,8 @@ class TextArtFragment : BaseFragment() {
     }
 
     private fun turnOnBluetooth() {
-        val btManager = requireContext().getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
-        val btAdapter = btManager.adapter
-        if (btAdapter.disable()) {
-            btAdapter.enable()
+        if (bluetoothManager.btAdapter.disable()) {
+            bluetoothManager.btAdapter.enable()
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt]: Create a function inside the singleton to enable bluetooth and shift everything there

File: app/src/main/java/org/fossasia/badgemagic/ui/fragments/SavedBadgesFragment.kt [hunk 4]
```diff
@@ -167,6 +170,28 @@ class SavedBadgesFragment : BaseFragment() {
...
+            Toast.makeText(context, R.string.bluetooth_enabled, Toast.LENGTH_SHORT).show()
+        }
+        builder.setNegativeButton("CANCEL") { _, _ ->
+            Toast.makeText(context, R.string.enable_bluetooth, Toast.LENGTH_SHORT).show()
+        }
+        builder.create().show()
+    }
+
+    private fun turnOnBluetooth() {
+        if (bluetoothManager.btAdapter.disable()) {
```
Comment (adityastic) [inline app/src/main/java/org/fossasia/badgemagic/ui/fragments/SavedBadgesFragment.kt]: Same here.

**Pull request merged**